src/ClassRegistrator_impl.hpp

```diff
@@ -70,8 +71,8 @@ namespace { // Implementation detail, so it's in an anon namespace
          *
          * @returns SEXP_UNDEF if everything went ok or a scheme exception if there was an error
          */
-        template<typename Res>
-        static sexp convert_or_exception(Chibi &chibi, sexp self, sexp convert_from, Res &set_result_to) {
+        template <typename Res>
+        static sexp convert_or_exception(Chibi &chibi, const sexp self, const sexp convert_from, Res &set_result_to) {
```

Comment:
Why `Chibi &chibi` is not const here?
And making arguments passed by value const is not really important.

Reply:
To use chibi as const some work is needed: it's methods are not const.

Reply:
Ok.